pages/using-uaa.md

```diff
@@ -116,92 +117,33 @@ The JHipster UAA server does three things out of the box:
-* It serves the JWT public key on "/oauth/token_key", which has to be consumed by all other microservices
+* It serves the JWT public key on `/oauth/token_key`, which has to be consumed by all other microservices
 
 The choices of a database, cache solution, search engine, build tools and further JHipster options are open to the developer.
 
-When a microservice boots up, it usually expects the UAA server is already up to share its public key. The service first calls "/oauth/token_key" to fetch the public key and configure it for key signing (`JwtAccessTokenConverter`)
+When a microservice boots up, it usually expects the UAA server is already up to share its public key. The service first calls `/oauth/token_key` to fetch the public key and configure it for key signing (`JwtAccessTokenConverter`).
+
+If the UAA is not up, the application will continue to start and fetch the public key at a later time.  There are two properties - `uaa.signature-verification.ttl` controls how long the key lives before it is fetched again, `uaa.signature-verification.public-key-refresh-rate-limit` limits requests to UAA to avoid spamming it. These values are usually left at their default values. In any case, if verification fails, then the microservice will check if there's a new key. That way, keys can be replaced on the UAA and the services will catch up.
 
 From this point there are two use cases that may happen in this basic setup: user calls and machine calls.
 
-For the user calls, the Angular application is requesting a bearer access token (which is a JWT token) using OAuth2's password grant, and saves this token to local storage (so no cookies are used at any stage). This bearer token is injected as authorization header into each request performed by the AngularJS `$http` object.
+For the user calls, a login request is sent to the gateway's `/auth/login` endpoint.  This endpoint uses `OAuth2TokenEndpointClientAdapter` to send a request to the UAA authenticating with the "password" grant.  Because this request happens on the gateway, the client ID and secret are not stored in any client-side code and are inaccessible to users.  The gateway returns a new Cookie containing the token, and this cookie is sent with each request performed from the client to the JHipster backend.
 
 For the machine calls, the machine has to authenticate as a UAA using client credentials grant. JHipster provides a standard solution, described in [secure inter-service-communication using feign clients](#inter-service-communication)
 
-### Common mistakes
-
-Here is a brief list of the very major things a developer should be aware of.
-
-#### ***The public key resolving leads to several issues***
-
-Using the token key endpoint of UAA makes it possible to easily change the signing key as often as needed. The need that a UAA server **must** be up in order to start a simple service, may cause problems in several use cases as testing or CI.
```

Comment:
I removed this section because apps no longer require the UAA to be up before starting.